content/en/docs/Contribution-guidelines/development-environment.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Development Environment"
+description: "How to set up a development environment"
+---
+
+
+### Introduction
+
+In this section it is explained how to set up a development environment:
+- As described in [Arquitecture](/docs/architecture/architecture), we would need to have the Kiali dependencies running in an Openshift or Kubernetes
+- We will use a port forward to access those services outside the cluster.
+- We will have the project source running locally. In this case we will set up an IDE.
+- Bookinfo application example will also be running on our cluster.
+
+![development_environment](/images/documentation/contribution/development_environment.png)
+
+### Prerrequisites 
+
+- Development tools are installed:
+  - [oc](https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html) or [kubectl](https://kubernetes.io/es/docs/tasks/tools/)
+  - [go](https://go.dev/)
+  - [make](https://www.gnu.org/software/make/)
+  - [npm](https://www.npmjs.com/)
+  - [yarn](https://yarnpkg.com/)
+- Istio and required services are running in Minikube or open shift. To install it following the above schema, it is possible to use the following scripts:
+  - `hack/istio/install-istio-via-istioctl.sh`: Installs the latest Istio release into *istio-system* namespace along with the Prometheus, Grafana, and Jaeger addons.
+  - `hack/istio/install-bookinfo-demo.sh`: Installs the Bookinfo demo that is found in the Istio release that was installed via the *hack/istio/install-istio-via-istioctl.sh* hack script.
+    - Pass in `-tg` to also install a traffic generator that will send messages periodically into the Bookinfo demo.
+    - If using Minikube and the `-tg` option, make sure you pass in the Minikube profile name via `-mp` if the profile name is not `minikube`.
+
+### Port forward
+
+Before the setup, we will need to do a port-forward of the services that kiali is using.
+
+We can use the `hack/run-kiali.sh` script for this purpose. To run it following the above schema:
+
+```shell
+./run-kiali.sh -pg 13000:3000 -pp 19090:9090 -app 8080 -es false -iu http://127.0.0.1:15014
```

Comment:
just a comment - reading this, it makes it seem as though the user NEEDS those options. I rarely pass any options and it works for me. Perhaps explain what those options do (and mention they are really optional).